cloud/keystone.go

```diff
@@ -243,6 +245,32 @@ func (client *keystoneV3Client) getTenant(filter func(*v3tenants.Project) bool)
 	return result, nil
 }
 
+func (client *keystoneV3Client) getDomain(filter func(domain *domains.Domain) bool) (*domains.Domain, error) {
+	opts := domains.ListOpts{}
```

Comment:
`ListOpts` has a `Enabled` filter. Shouldn't we be interested only in enabled domains?

Reply:
Good idea. Added to getTenants too.